torch/csrc/jit/codegen/cuda/test/test_gpu_rng.cu

```diff
@@ -65,7 +65,47 @@ __global__ void generate_uniform_kernel(
   }
 }
 
-at::Tensor generate_uniform(int64_t size, at::ScalarType dtype) {
+template <typename T>
+__global__ void generate_normal_kernel(
```

Comment:
Looks like now this kernel and `generate_uniform_kernel` are only different in `curand_uniform*` vs `curand_normal*`. Could we merge these two kernels into one by adding a flag to switch between uniform and normal?

Reply:
I had the same thought. Will try to simplify the test with a template function.